warehouse/ingest-core/src/main/java/datawave/ingest/data/config/ingest/BaseIngestHelper.java

```diff
@@ -171,8 +178,37 @@ public abstract class BaseIngestHelper extends AbstractIngestHelper implements C
     private CompositeIngest compositeIngest;
     private VirtualIngest virtualIngest;
 
+    protected String ageoffDateField = null;
+
     protected boolean useMostPreciseFieldTypeRegex = false;
 
+    @Override
+    public long getAgeOffDate(RawRecordContainer record, Multimap<String,NormalizedContentInterface> fields) {
+        // if the record already provides the ageoff, then return its value
+        if (record instanceof WithAgeOff) {
+            return ((WithAgeOff) record).getAgeOffDate();
+        }
+        // otherwise lets try and pull it from one of the fields
+        long ageOffDate = record.getDate();
+        if (ageoffDateField != null && !ageoffDateField.isBlank()) {
+            Collection<NormalizedContentInterface> values = fields.get(ageoffDateField);
+            if (values != null && !values.isEmpty()) {
+                // take the largest date
+                for (NormalizedContentInterface value : values) {
+                    try {
+                        Date d = new DateNormalizer().denormalize(value.getIndexedFieldValue());
```

Comment:
Should I configure a separate date format like we do for the event date in the AbstractEventRecordReader ?

Reply:
not sure that's necessary

Reply:
OBE